redash/tasks/general.py

```diff
@@ -60,3 +62,33 @@ def send_mail(to, subject, html, text):
 
 def sync_user_details():
     users.sync_last_active_at()
+
+
+def purge_failed_jobs():
+    jobs = rq_redis_connection.scan_iter('rq:job:*')
+
+    is_idle = lambda key: rq_redis_connection.object('idletime', key) > settings.JOB_DEFAULT_FAILURE_TTL
+    has_failed = lambda key: rq_redis_connection.hget(key, 'status') == b'failed'
+
+    def not_in_any_failed_registry(key):
+        """ This function should reject any key which is inside any FailedJobRegistry.
+        However, at the moment on RQ v1.1 the @job decorator does not allow setting of a failure_ttl, so jobs
+        are kept inside the FailedJobRegistry for a year and there is no easy way tweak that.
+        This has already been fixed on rq/master (https://github.com/rq/rq/pull/1130) and will be available on the
+        next release. Until then, we simply don't reject keys in a FailedJobRegistry and purge any failed jobs
+        that have been idle for over settings.JOB_DEFAULT_FAILURE_TTL.
+        Once a new RQ release is available, we can delete this comment and the following line:"""
+        return True
+
+        with Connection(rq_redis_connection):
+            failed_registries = [FailedJobRegistry(queue=q) for q in Queue.all()]
+
+        job_id = lambda key: key.decode().split(':').pop()
+        return all([job_id(key) not in registry for registry in failed_registries])
+
+    stale_jobs = [key for key in jobs if is_idle(key) and has_failed(key) and not_in_any_failed_registry(key)]
+
+    for key in stale_jobs:
+        rq_redis_connection.delete(key)
```

Comment:
Maybe worth removing it from the FailedRegistry while we at it?

Reply:
We _could_ do that, but the point is that we want to let the FailedJobRegistry handle its own state. From what I can tell, there aren't any dire consequences to have ghost job ids in the FailedJobRegistry (it is only used for requeueing and in that case - these jobs will simply not get requeued)

Reply:
If we delete from `FailedJobRegistry`, we might as well just do that and avoid checking for job inclusion (and avoid the whole comment+bypass at the top of the function).

🤔

Reply:
I'm just worried that over a year it can accumulate quite a lot of job ids there, which might have some consequences in performance or at least memory usage.

Re. avoid checking job inclusion: I guess we can skip this.

Reply:
Yeah 02555b9 makes things simpler.